polaris-react/src/components/Tooltip/Tooltip.tsx

```diff
@@ -158,7 +158,13 @@ export function Tooltip({
 
   const handleKeyUp = useCallback(
     (event: React.KeyboardEvent) => {
-      if (event.key !== 'Escape') return;
+      if (
+        event.key !== 'Escape' &&
+        event.key !== 'Enter' &&
+        event.key !== ' '
+      ) {
+        return;
+      }
       handleClose?.();
```

Comment:
Updates the `Tooltip` to dismiss on enter/space key interactions (matching the behavior `onClick`)

https://github.com/Shopify/polaris/assets/32409546/4c576622-3bc8-43b4-9ec9-f876cf98d7c1